### PR TITLE
Publish "nobridge" version of the library

### LINKED ISSUE #1739: Unable to mock GHObject#id
Hi, 

it seems like it's impossible to mock `GHObject` via Mockito, I observe this behavior:

```
try {
    var a = Mockito.mock(GHObject.class);
    when(a.getId()).thenReturn(123L);
} catch (Throwable e) {
    e.printStackTrace();
}
```

The exception:
```
java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.concurrent.Callable.call()" is null
	at org.kohsuke.github.GHObject.getId(GHObject.java:140)
	at org.kohsuke.github.GHObject.getId(GHObject.java:140)
```

I believe this may be related to bridge methods, but I have no idea how to implement a workaround.

Could you please elaborate if this is possible?

### PR DESCRIPTION
**Pull request by bitwiseman**
# Description

Converts the project to a multi-module structure  and adds a new "No Bridge" build output.  

* Multi-module - allows us to produce more than one flavor of library based on the same code.
* "No Bridge" build flavor - compiles the library without generating bridge methods. This flavor of the library does NOT guarantee binary backward compatibility. However, it does unblock mocking ( fixes #1739 ). 

Multi-module will also allow us to begin producing a v2 beta flavor without breaking the consumers of the v1 line. 

# Before submitting a PR:

- [ ] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [ ] Add JavaDocs and other comments explaining the behavior. 
- [ ] When adding or updating methods that fetch entities, add `@link` JavaDoc entries to the relevant documentation on https://docs.github.com/en/rest . 
- [ ] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [ ] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [ ] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [ ] Fill in the "Description" above with clear summary of the changes. This includes:
  - [ ] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
  - [ ] Provide links to relevant documentation on https://docs.github.com/en/rest where possible. If not including links, explain why not.
- [ ] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [ ] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by bitwiseman** [comment]
@gsmet @Haarolean 
Do you think you could try this out and see if it helps with the mocking issue?

**Comment by Haarolean** [comment]
@bitwiseman sure, thank you! Will take a look

File: github-api-nobridge/pom.xml [hunk 1]
```diff
@@ -0,0 +1,306 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.kohsuke</groupId>
```
Comment (Haarolean) [inline github-api-nobridge/pom.xml:3]: groupId declaration is redundant here due to the module being inherited from a parent

**Comment by Haarolean** [comment]
@bitwiseman I can confirm that mocking objects works now. Thank you!

**Comment by bitwiseman** [comment]
I'm going to make this a `release/v1.x-nobridge` branch instead of multimodule.

**Pull request closed without merging**